file/writable_file_writer.cc

```diff
@@ -216,9 +216,21 @@ IOStatus WritableFileWriter::Flush() {
   return s;
 }
 
+std::string WritableFileWriter::GetFileChecksum() {
+  if (checksum_generator_ != nullptr) {
+    if (!checksum_finalized_) {
+      checksum_generator_->Finalize();
```

Comment:
Only calling Finalize() inside GetFileChecksum() defeats the purpose of Finalize(). The application can do it if they want to. In order for Finalize() to make sense, we should call it when we close the file. Otherwise, there is no point of having such a function.

Reply:
@siying I see the logic here. Will move to Close and update the GetChecksum order in builder.cc